tests/telemetry/test_writer.py

```diff
@@ -189,6 +193,114 @@ def test_telemetry_graceful_shutdown(telemetry_writer, test_agent_session):
     assert events[1]["request_type"] == "app-started"
 
 
+def test_send_tracers_count_metric(mock_time, telemetry_writer, test_agent_session):
+    telemetry_writer.add_count_metric(TELEMETRY_TRACER, "test-metric", 1, {"hi": "HELLO", "NAME": "CANDY"})
+    telemetry_writer.add_count_metric(TELEMETRY_TRACER, "test-metric", 1, {})
```

Comment:
These two data points should not be combined. This was a bug in my initial implementation. Tags should be associated with individual data points not metric names. 

We should only combine datapoints into one series object if they contain the same tags. For now we should avoid combing metrics and this optimization in a follow up PR

Reply:
Thanks @mabdinur! PR updated with this comment :wink: